modules/raft/src/main/java/org/apache/ignite/internal/raft/server/RaftGroupOptions.java

```diff
@@ -194,4 +205,40 @@ public RaftGroupOptions serverDataPath(Path serverDataPath) {
         return this;
     }
 
+    /**
+     * Externally established peers.
+     *
+     * <p>The idea is that, if a Raft group was forcefully repaired (because it lost majority) with {@link Node#resetPeers(Configuration)},
+     * the old majority nodes might come back online. If this happens and we do nothing, they might elect a leader from the old majority
+     * that could hijack leadership and cause havoc in the repaired group.
+     *
+     * <p>To prevent this, on a starup, current voting set (aka peers) of the repaired group may be set via
+     * {@link NodeOptions#getExternallyEstablishedPeers()}. If a Raft node sees at startup that, according to its log it's a member of
+     * the voting set, but not a member of externally established peers, it realizes that it should abstain from becoming a leader to avoid
+     * a hijack.
+     *
+     * <p>If the group is reconfigured according to the Raft protocol (that is, non-forcefully), a Raft node might see that it's not
+     * a member of the externally established peers anymore and start abstaining from becoming a leader as well, but that's not a problem
```

Comment:
In case of graceful MG topology update what will happen first inner MG topology update or CMG.MGTopology field update?

Reply:
The approach has been changed and we only modify `NodeOptions` (adding new behavior) if an actual repair happened, so this will not affect graceful MG reconfiguration anymore